src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -108,40 +109,55 @@ private static void EncodeToUtf16_Ssse3(ReadOnlySpan<byte> bytes, Span<char> cha
                                  (byte)'8', (byte)'9', (byte)'a', (byte)'b',
                                  (byte)'c', (byte)'d', (byte)'e', (byte)'f');
 
+
+            nuint lengthSubVector128 = (nuint)bytes.Length - (nuint)Vector128<int>.Count;
             do
             {
                 // Read 32bits from "bytes" span at "pos" offset
                 uint block = Unsafe.ReadUnaligned<uint>(
                     ref Unsafe.Add(ref MemoryMarshal.GetReference(bytes), pos));
 
+                // TODO: Remove once cross-platform Shuffle is landed
+                // https://github.com/dotnet/runtime/issues/63331
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
+                static Vector128<byte> Shuffle(Vector128<byte> value, Vector128<byte> mask) =>
+                    Ssse3.IsSupported ? Ssse3.Shuffle(value, mask) : AdvSimd.Arm64.VectorTableLookup(value, mask);
+
                 // Calculate nibbles
-                Vector128<byte> lowNibbles = Ssse3.Shuffle(
+                Vector128<byte> lowNibbles = Shuffle(
                     Vector128.CreateScalarUnsafe(block).AsByte(), shuffleMask);
-                Vector128<byte> highNibbles = Sse2.ShiftRightLogical(
-                    Sse2.ShiftRightLogical128BitLane(lowNibbles, 2).AsInt32(), 4).AsByte();
+
+                // ExtractVector128 is not entirely the same as ShiftRightLogical128BitLane, but it works here since
+                // first two bytes in lowNibbles are guaranteed to be zeros
+                Vector128<byte> shifted = Sse2.IsSupported ?
+                    Sse2.ShiftRightLogical128BitLane(lowNibbles, 2) :
+                    AdvSimd.ExtractVector128(lowNibbles, lowNibbles, 2);
+
+                Vector128<byte> highNibbles = Vector128.ShiftRightLogical(shifted.AsInt32(), 4).AsByte();
 
                 // Lookup the hex values at the positions of the indices
-                Vector128<byte> indices = Sse2.And(
-                    Sse2.Or(lowNibbles, highNibbles), Vector128.Create((byte)0xF));
-                Vector128<byte> hex = Ssse3.Shuffle(asciiTable, indices);
+                Vector128<byte> indices = (lowNibbles | highNibbles) & Vector128.Create((byte)0xF);
+                Vector128<byte> hex = Shuffle(asciiTable, indices);
 
                 // The high bytes (0x00) of the chars have also been converted
                 // to ascii hex '0', so clear them out.
-                hex = Sse2.And(hex, Vector128.Create((ushort)0xFF).AsByte());
+                hex &= Vector128.Create((ushort)0xFF).AsByte();
 
                 // Save to "chars" at pos*2 offset
                 Unsafe.WriteUnaligned(
                     ref Unsafe.As<char, byte>(
                         ref Unsafe.Add(ref MemoryMarshal.GetReference(chars), pos * 2)), hex);
```

Comment:
Couldn't this be `hex.StoreUnsafe(ref charsBase, pos * 2)` where you pre-extract `ref ushort charsBase = ref Unsafe.As<char, ushort>(ref MemoryMarshal.GetReference(chars))`

Would simplify the amount of code in the loop (less work for the JIT) and make it a bit more readable